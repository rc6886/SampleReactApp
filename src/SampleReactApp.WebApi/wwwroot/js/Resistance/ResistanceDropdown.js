import React from 'react';
import { Dropdown } from 'semantic-ui-react';

import BandColors from './BandColors';

export const toDropdownOptions = (bandColors) => {
  return Object.keys(BandColors).map(color => ({
    text: color,
    value: color,
  }));
};

const ResistanceDropdown = (props) => (
  <Dropdown
    fluid
    selection
    placeholder={props.placeholder}
    options={props.options}
  />
);

export default ResistanceDropdown;
