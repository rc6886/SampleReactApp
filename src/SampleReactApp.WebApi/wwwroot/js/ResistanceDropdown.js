import React from 'react';
import { Dropdown } from 'semantic-ui-react';

const ResistanceDropdown = (props) => (
  <Dropdown
    fluid
    selection
    placeholder={props.placeholder}
    options={props.options}
  />
);

export default ResistanceDropdown;
