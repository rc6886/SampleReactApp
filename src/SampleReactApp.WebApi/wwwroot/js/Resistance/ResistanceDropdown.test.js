import React from 'react';
import { shallow } from 'enzyme';

import ResistanceDropdown from './ResistanceDropdown';

describe('Resistance Dropdown', () => {
  const props = {
    placeholder: 'Dropdown 1',
    options: [
      {
        text: 'Value 1',
        value: 'Value1',
      },
      {
        text: 'Value 2',
        value: 'Value2',
      },
    ],
  };

  const dropdown = <ResistanceDropdown {...props} />;

  it('Should Render Dropdown', () => {
    const wrapper = shallow(dropdown);
    expect(wrapper).toMatchSnapshot();
  });
});