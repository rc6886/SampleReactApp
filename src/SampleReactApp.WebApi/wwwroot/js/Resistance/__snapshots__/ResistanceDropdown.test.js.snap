// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resistance Dropdown Should Render Dropdown 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  deburr={false}
  fluid={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  openOnFocus={true}
  options={
    Array [
      Object {
        "text": "Value 1",
        "value": "Value1",
      },
      Object {
        "text": "Value 2",
        "value": "Value2",
      },
    ]
  }
  placeholder="Dropdown 1"
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  selection={true}
  wrapSelection={true}
/>
`;
